django/db/backends/utils.py

```diff
@@ -125,9 +125,10 @@ def typecast_time(s):  # does NOT store time zone information
     hour, minutes, seconds = s.split(':')
     if '.' in seconds:  # check whether seconds have a fractional part
         seconds, microseconds = seconds.split('.')
+        microseconds = microseconds.ljust(6, '0')
```

Comment:
Thanks for pointing it out @adamchainz. I think it will be better for consistency and performance to do it in the same way as `typecast_timestamp`
`int((microseconds + '000000')[:6])`